harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/AbstractHarvestError.java

```diff
@@ -0,0 +1,24 @@
+package org.fao.geonet.kernel.harvest.harvester;
+
+import org.fao.geonet.monitor.harvest.AbstractHarvesterErrorCounter;
+
+import com.yammer.metrics.core.Counter;
+
+import jeeves.monitor.MonitorManager;
+import jeeves.server.context.ServiceContext;
+
+public class AbstractHarvestError {
+	
+	public AbstractHarvestError(ServiceContext context) {
+		super();
+		// we don't catch anything because this is not critical
+		// if the following code is not executed, we just want to
+		// avoid to raise an exception.
+		try {
+			MonitorManager mm = context.getBean(MonitorManager.class);
```

Comment:
> so could we replace that ServiceContext by ApplicationContextHolder.get().getBean instead ?

sure, but the MonitorManager is also instanciated by Jeeves ( https://github.com/geonetwork/core-geonetwork/blob/develop/core/src/main/java/jeeves/server/JeevesEngine.java#L319-L324 ) ; it might be interesting to reconsider the metrics introduction into GN completely.

> or even use @autowired on MonitorManager

I am not very fond of having a dangling reference to the MonitorManager onto the *HarvestError class. I think it is better to keep it as simple as it is currently. And *HarvestError objects are not spring-managed beans, they are dynamically created at runtime, so there is no bean injection possible as far as I know